test/integration/host_js_test.rb

```diff
@@ -25,8 +25,8 @@ class HostJSTest < IntegrationTestWithJavascript
     test 'class parameters and overrides are displayed correctly for strings' do
       host = FactoryGirl.create(:host, :with_puppetclass)
       FactoryGirl.create(:puppetclass_lookup_key, :as_smart_class_param, :with_override,
-                                      :key_type => 'string', :default_value => true, :path => "fqdn",
-                                      :puppetclass => host.puppetclasses.first, :overrides => {host.lookup_value_matcher => false})
+                                      :key_type => 'string', :default_value => 'true', :path => "fqdn",
```

Comment:
I don't understand these changes nor the changes to unit tests. The model still takes in booleans, integers, arrays, etc.. it's just the API controllers should only take strings. Why changing the unit tests to test only strings? At the model level we don't need to cast every time.